src/scripts/index.js

```diff
@@ -0,0 +1,302 @@
+
+function getCardElement(data) {
+  const cardElement = cardTemplate.content
+    .querySelector(".card")
+    .cloneNode(true);
+  const cardImage = cardElement.querySelector(".card__image");
+  const cardTitle = cardElement.querySelector(".card__title");
+  const cardLikeBtn = cardElement.querySelector(".card__like-btn");
+  const cardDeleteBtn = cardElement.querySelector(".card__trash-btn");
+
+  cardImage.src = data.link;
+  cardImage.alt = data.name;
+  cardTitle.textContent = data.name;
+
+  cardImage.addEventListener("click", () => {
+    previewModalImageEl.src = data.link;
+    previewModalImageEl.alt = data.name;
+    previewModalCaptionEl.textContent = data.name;
+
+    openModal(previewModal);
+  });
+
+  cardLikeBtn.addEventListener("click", (evt) => {
+    api
+      .changeLikeStatus(data._id, data.isLiked)
+      .then((updatedCard) => {
+        console.log(updatedCard);
+        evt.target.classList.toggle("card__like-btn_liked");
+      })
+      .catch((err) => console.error(`Error updating like status: ${err}`));
+  });
+
+  cardDeleteBtn.addEventListener("click", () => {
+    selectedCard = cardElement;
+    selectedCardID = data._id;
+    openModal(deleteModal);
+  });
+
+  return cardElement;
+}
+
+function handleEscClose(evt) {
+  if (evt.key === "Escape") {
+    const openModal = document.querySelector(".modal_opened");
+    if (openModal) {
+      closeModal(openModal);
+    }
+  }
+}
+
+function openModal(modal) {
+  if (!modal.classList.contains("modal_opened")) {
+    modal.classList.add("modal_opened");
+    document.addEventListener("keydown", handleEscClose);
+  }
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscClose);
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+
+  const submitBtn = evt.submitter;
+  setButtontext(submitBtn, true, "Save", "Saving...");
+
+  // Call API to update user info
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch((error) => {
+      console.error("Error updating profile:", error);
+    })
+    .finally(() => {
+      setButtontext(submitBtn, false, "Save", "Saving...");
+    });
+}
+
+//TODO - implement loading text for all other form submissions
+
+function handleAddCardSubmit(evt) {
+  evt.preventDefault();
+  const submitBtn = evt.submitter;
+
+  setButtontext(submitBtn, true, "Save", "Saving...");
+
+  const cardData = {
+    name: cardNameInput.value,
+    link: cardLinkInput.value,
+  };
+
+  api
+    .addCard(cardData)
+    .then((newCard) => {
+      cardsList.prepend(getCardElement(newCard)); // Prepend new card
+      cardForm.reset(); // Reset form
+      resetValidation(cardForm, settings); // Reset validation
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting. You need to call `disableButton` there.